src/todo_final.ts

```diff
@@ -44,32 +45,50 @@ function completeList() {
 	for (const item of completedItems) {
 		console.log(item.id + " " + item.text);
 	}
+	return completedItems;
+}
+
+// mark a do item as done
+// find returns an object filter returns an array
+function markTodoAsDone(userId: number) {
+	console.log(todoItems);
+	for (const item of todoItems) {
+		// check if user input matches or not
+		if (item.id === userId) {
+			item.checked = true;
+			// when it matches return this
+			return "Successful";
+		}
+	}
+	// after looping through each item then print this
+	return "Error, Please provide item from the list";
 }
 
 // list uncompleted todos
 function uncompletedList() {
 	/*
 	 * filter is an array method that returns the position of an element in the array
 	 */
-
 	const uncompletedItems = todoItems.filter((item) => item.checked === false);
 	for (const item of uncompletedItems) {
 		console.log(item.id + " " + item.text);
 	}
+	return uncompletedItems;
 }
+
 // for user to write the text of the item they want to remove since they dont know abou the id
 function findTodoByText(text: string) {
-	// find the text that matches the text  inside the array
+	// find the index where the  text that matches the text  inside the array
 	const findTodoWithText = todoItems.findIndex(
 		(todoInsideArray) => todoInsideArray.text === text
 	);
-	return todoItems[findTodoWithText]; //return the item not the text itself
+	return todoItems[findTodoWithText]; // return the item not the text itself
 }
 
 // remove todo by text
 
 function removeTodoByText(text: string) {
-	removeTodo(findTodoByText(text));
+	return removeTodo(findTodoByText(text));
 }
 
 function removeTodo(itemsToRemove: any[number]) {
```

Comment:
This function only removes one item, so its argument should be `itemToRemove`, not `itemsToRemove`. Also, the `any[number]` type is wrong. Looking at the code, I can see that this function is designed to take a `TodoItem` as its argument, not a number or an array.